rdflib/graph.py

```diff
@@ -1983,38 +2003,104 @@ def graph(self, identifier=None, base=None):
         g = self._graph(identifier)
         g.base = base
 
-        self.store.add_graph(g)
+        self.store.add_graph(identifier)
         return g
 
     def parse(
         self,
-        source=None,
-        publicID=None,
-        format=None,
-        location=None,
-        file=None,
-        data=None,
+        source: Optional[
+            Union[IO[bytes], TextIO, InputSource, str, bytes, pathlib.PurePath]
+        ] = None,
+        publicID: Optional[str] = None,
+        format: Optional[str] = None,
+        location: Optional[str] = None,
+        file: Optional[Union[BinaryIO, TextIO]] = None,
+        data: Optional[Union[str, bytes]] = None,
         **args,
     ):
-        c = ConjunctiveGraph.parse(
-            self, source, publicID, format, location, file, data, **args
+        """
+        Parse source adding the resulting triples to its own context
+        (sub graph of this graph).
+
+        See :meth:`rdflib.graph.Graph.parse` for documentation on arguments.
+
+        :Returns:
+
+        The graph into which the source was parsed. In the case of n3
+        it returns the root context.
+        """
+
+        quadformats = ["trig", "trix", "nquads", "hext", "json-ld"]
```

Comment:
I think this is meant to solve the problem of dataset serialization formats that send the default graph to the graph matching the "publicID" (whatever a "public ID" is), but keeping a list of formats RDFLib happens to know have a concept of "quads" or "graphs" seems a very poor way to handle that. More importantly, I think handling *that* problem can be done separately from just changing the store API.

Reply:
Yeah, what _does_ `publicID` actually mean? Is it just alternate (archaic?) terminology for the graph identifier, i which case we should probably just purge it, or does it have some other meaning?

Reply:
> Yeah, what _does_ `publicID` actually mean? Is it just alternate (archaic?) terminology for the graph identifier, i which case we should probably just purge it, or does it have some other meaning?

From the docstring: “The publicID argument is for specifying the logical URI for the case that it's different from the physical source URI”.

It's necessary for correct serialization, e.g. reading the file `examples/foaf.rdf` and parsing it as `data` without a publicID results in URLs being rewritten as the file source, e.g.
```py
rdflib.term.URIRef('file:///home/gjh/dev/rdflib/images/timbl-image-by-Coz-cropped.jpg')
```
whereas using the publicID of `http://dig.csail.mit.edu/2008/webdav/timbl/foaf.rdf` results in the correct serialisation:
```py
rdflib.term.URIRef('http://dig.csail.mit.edu/2008/webdav/timbl/images/timbl-image-by-Coz-cropped.jpg')
```
Just using the CSAIL URL as identifier doesn't result in correct serialization so it does appear that `publicID` has a valid and continued role.

Reply:
OK, ok, I think I knew that or should have worked that out! So `publicID` stays as is.